features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/lwip_sys_arch.c

```diff
@@ -327,15 +377,10 @@ void sys_sem_free(sys_sem_t *sem) {}
  * @param mutex pointer to the mutex to create
  * @return a new mutex */
 err_t sys_mutex_new(sys_mutex_t *mutex) {
-#ifdef CMSIS_OS_RTX
-#if defined(__MBED_CMSIS_RTOS_CA9) || defined(__MBED_CMSIS_RTOS_CM)
-    memset(mutex->data, 0, sizeof(int32_t)*4);
-#else
-    memset(mutex->data, 0, sizeof(int32_t)*3);
-#endif
-    mutex->def.mutex = mutex->data;
-#endif
-    mutex->id = osMutexCreate(&mutex->def);
+    memset(&mutex->data, 0, sizeof(mutex->data));
```

Comment:
Some of the mutex attributes are not set in this function. It might be safer to memset the entire mutex to 0 rather than just the data.